drivers/ieee802154/ieee802154_gecko.c

```diff
@@ -0,0 +1,588 @@
+	if (status != RAIL_STATUS_NO_ERROR) {
+		LOG_DBG("RAIL_IEEE802154_Init returned an error %d", status);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int ieee802154_gecko_init(struct device *dev)
+{
+	struct ieee802154_gecko_context *dev_data = DEV_DATA(dev);
+	int ret;
+
+	k_sem_init(&dev_data->tx_wait, 0, 1);
+
+	ret = ieee802154_gecko_init_rail(dev);
+	if (ret != 0) {
+		LOG_ERR("%d: Failed to initialize RAIL", ret);
+		return ret;
+	}
+
+	ieee802154_gecko_irq_config();
+
+	ret = ieee802154_gecko_init_ieee802154(dev);
+	if (ret != 0) {
+		LOG_ERR("%d: Failed to initialize IEEE 802.15.4 Radio", ret);
+		return ret;
+	}
+
+	LOG_INF("Device %s initialized", DEV_NAME(dev));
+
+	return 0;
+}
+
+static int ieee802154_gecko_configure(struct device *dev,
+				      enum ieee802154_config_type type,
+				      const struct ieee802154_config *config)
+{
+	return 0;
+}
+
+static struct ieee802154_radio_api ieee802154_gecko_radio_api = {
+	.iface_api.init = ieee802154_gecko_iface_init,
+
+	.get_capabilities = ieee802154_gecko_get_capabilities,
+	.cca = ieee802154_gecko_cca,
+	.set_channel = ieee802154_gecko_set_channel,
+	.filter = ieee802154_gecko_filter,
+	.set_txpower = ieee802154_gecko_set_txpower,
+	.start = ieee802154_gecko_start,
+	.stop = ieee802154_gecko_stop,
+	.tx = ieee802154_gecko_tx,
+	.configure = ieee802154_gecko_configure,
+};
+
+DEVICE_DECLARE(ieee802154_gecko_dev0);
+
+static struct ieee802154_gecko_context ieee802154_gecko_dev0_data;
+static const struct ieee802154_gecko_dev_cfg ieee802154_gecko_dev0_config = {
+	.init_tx_power_level_raw = CONFIG_IEEE802154_GECKO_TXPOWER_RAW,
+};
+
+static void ieee802154_gecko_rail_cb(RAIL_Handle_t railHandle,
+				     RAIL_Events_t event)
+{
+	struct device *dev = DEVICE_GET(ieee802154_gecko_dev0);
+
+	if (event & RAIL_EVENT_CAL_NEEDED) {
+		RAIL_Calibrate(railHandle, NULL, RAIL_CAL_ALL_PENDING);
+	}
+
+	if (event & (RAIL_EVENT_TX_ABORTED |
+		     RAIL_EVENT_TX_BLOCKED |
+		     RAIL_EVENT_TX_UNDERFLOW |
+		     RAIL_EVENT_TX_CHANNEL_BUSY)) {
+		LOG_DBG("RAIL_Events_t 0x%llx", event);
+		ieee802154_gecko_dev0_data.tx_status = -EIO;
+		k_sem_give(&ieee802154_gecko_dev0_data.tx_wait);
+	}
+
+	if (event & RAIL_EVENT_TX_PACKET_SENT) {
+		LOG_DBG("RAIL_Events_t: TX_PACKET_SENT");
+		ieee802154_gecko_dev0_data.tx_status = 0;
+		k_sem_give(&ieee802154_gecko_dev0_data.tx_wait);
+	}
+
+	if (event & RAIL_EVENT_RX_PACKET_RECEIVED) {
+		RAIL_RxPacketInfo_t packet_info;
+		RAIL_RxPacketHandle_t rx_packet_handle;
+
+		LOG_DBG("RAIL_Events_t: RX_PACKET_RECEIVED");
+
+		rx_packet_handle = RAIL_GetRxPacketInfo(
+				ieee802154_gecko_dev0_data.rail_handle,
+				RAIL_RX_PACKET_HANDLE_NEWEST,
+				&packet_info);
+		if ((rx_packet_handle != RAIL_RX_PACKET_HANDLE_INVALID) &&
+		    (packet_info.packetStatus == RAIL_RX_PACKET_READY_SUCCESS)) {
+			ieee802154_gecko_rx(dev, rx_packet_handle, &packet_info);
+		}
+	}
+}
+
+#define L2 IEEE802154_L2
```

Comment:
Adding Thread configuration

```suggestion
#if defined(CONFIG_NET_L2_IEEE802154)
#define L2 IEEE802154_L2
#define L2_CTX_TYPE NET_L2_GET_CTX_TYPE(IEEE802154_L2)
#define MTU 125
#elif defined(CONFIG_NET_L2_OPENTHREAD)
#define L2 OPENTHREAD_L2
#define L2_CTX_TYPE NET_L2_GET_CTX_TYPE(OPENTHREAD_L2)
#define MTU 1280
#endif
```